app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsAdvancedScreen.kt

```diff
@@ -345,6 +350,28 @@ object SettingsAdvancedScreen : SearchableSettings {
                     pref = basePreferences.alwaysUseSSIVToDecode(),
                     title = stringResource(MR.strings.pref_always_use_ssiv_to_decode),
                 ),
+                Preference.PreferenceItem.EditTextPreference(
+                    pref = maxBitmapSize,
+                    title = stringResource(MR.strings.pref_max_bitmap_size),
+                    subtitle = stringResource(MR.strings.pref_max_bitmap_size_summary),
+                    onValueChanged = {
+                        if (it.isDigitsOnly() && it <= maxBitmapSize.defaultValue()) {
+                            context.toast(MR.strings.pref_max_bitmap_size_success)
+                        } else {
+                            context.toast(MR.strings.pref_max_bitmap_size_error)
+                            return@EditTextPreference false
+                        }
+                        true
+                    },
+                ),
+                Preference.PreferenceItem.TextPreference(
+                    title = stringResource(MR.strings.pref_max_bitmap_size_reset),
+                    enabled = remember(maxBitmap) { maxBitmap != maxBitmapSize.defaultValue() },
+                    onClick = {
+                        maxBitmapSize.delete()
+                        context.toast(MR.strings.pref_max_bitmap_size_success)
+                    },
+                ),
```

Comment:
I'm not sure how i feel about an arbitrary value

Reply:
@AntsyLich Which arbitrary value are you referring to? The default value is set as GLUtil.maxTextureSize when declaring the function in BasePreferences, so mantain the way it currently works for non-affected users (which are the majority of the userbase).

I took this approach precisely because I don't want to be hardcoding a value that might not cover all cases or that might cover more than needed. This would only allow the user to lower the threshold in the event that maxTextureSize results too permissive in practice, so unless someone changes this value, the decoder will work just as it does currently.

Reply:
I'd say use a slider with limited values, like 4k, 8k, 16k, etc.